lein/helpers/src/helpers/core.clj

```diff
@@ -0,0 +1,44 @@
+(ns helpers.core
+  (:require [leiningen.pom :as pom]
+            [clojure.java.io :as io]
+            [rewrite-clj.zip :as zip]
+            [clojure.data.json :as json]
+            [leiningen.core.project :as project])
+  (:gen-class))
+
+(defn- match? [dep-name dep-version {:keys [dependency previous]}]
+  (and (= previous dep-version)
+       (or (= dependency dep-name)
+           (= dependency (str dep-name "/" dep-name)))))
+
+(defn- update-dependency [dependencies dep]
+  (let [dep-name (-> dep (zip/get 0) zip/sexpr str)
+        dep-version (-> dep (zip/get 1) zip/sexpr str)]
+    (if-let [updated (->> dependencies
+                          (filter (partial match? dep-name dep-version))
+                          (first))]
+      (zip/assoc dep 1 (:version updated))
+      dep)))
+
+(defn update-dependencies [{:keys [file dependencies]}]
+  (-> file
+      (zip/of-string)
+      (zip/find-value zip/next 'defproject)
+      (zip/find-value :dependencies)
+      (zip/right)
+      (#(zip/map (partial update-dependency dependencies) %))
+      (zip/root-string)))
+
+(defn generate-pom [{:keys [file]}]
+  (let [proj (project/read-raw (io/reader (char-array file)))]
```

Comment:
Will this execute code that lives in the `project.clj` file? I think so? Not sure if that's safe in this context or not.

https://github.com/technomancy/leiningen/blob/830d19d5b9f9e2b0e3411428d83422791ca2f814/leiningen-core/src/leiningen/core/project.clj#L1094-L1110

Reply:
I think that's mostly fine, as long as the code that's executed is from the source repository. The same thing is possible in a `setup.py`, `Gemfile` or `mix.exs` file. My main worry would be if an external package would be able to execute code in this context, but I don't think that's the case?

Reply:
That's correct, downloading POMs (Maven/Clojure dependency manifest) and JARs (dependency files) doesn't execute any code. Only the code that exists in the project would be executed.